### ISSUE TITLE
ReturnCount gives false positives

### ISSUE THREAD

**Issue by ham1** [checkstyle/checkstyle#5305]
Sorry for not filling in all of the info, I don't have time at the moment, i'm hoping the issue is obvious to track down, if not then I will try to provide more info.

I'm trying to add some more checkstyle rules to JMeter, I've added the following
```xml
    <module name="ReturnCount">
      <property name="max" value="10"/>
    </module>
```

However when I run it I get, among others:
```
[checkstyle] [ERROR] /home/td/coding/jmeter/src/jorphan/org/apache/jorphan/gui/RateRenderer.java:41:5: Return count is 2 (max allowed is 1). [ReturnCount]
[checkstyle] [ERROR] /home/td/coding/jmeter/src/jorphan/org/apache/jorphan/util/Converter.java:44:5: Return count is 13 (max allowed is 10). [ReturnCount]
```

This happens with checkstyle-all jar versions 8.{2,3,4,5}.

I'd not expect "Return count is 2 (max allowed is 1)" when I've set it to 10.

---

`/var/tmp $ javac YOUR_FILE.java`:

```
[[PLACE YOUR OUTPUT HERE]]
```

`/var/tmp $ cat YOUR_FILE.java`:

```
[[PLACE YOU OUTPUT HERE]]
```

`/var/tmp $ cat config.xml`:

```
[[PLACE YOUR OUTPUT HERE]]
```

`/var/tmp $ java -Duser.language=en -Duser.country=US -jar checkstyle-X.XX-all.jar -c config.xml YOUR_FILE.java`:

```
[[PLACE YOUR OUTPUT HERE]]
```

---------------

Describe what you expect in detail.

--------------

Still not clear ??? - see example - http://checkstyle.sourceforge.net/report_issue.html#How_to_report_a_bug

**Comment by romani** [comment]
issue is far from obvious , please provide all details

> I'd not expect "Return count is 2 (max allowed is 1)" when I've set it to 10.

it is clear misconfiguration on your side.

**Comment by ham1** [comment]
Thank you for the quick reply!

Just to clarify, what config are you referring to?
I've set max to 10 in our `checkstyle.xml`, as above fragment. Then in the same run I get:
>Return count is 13 (max allowed is 10)
And
>Return count is 2 (max allowed is 1)

It's not obvious why it would be picking up my setting for some files and not other.

I will try to get a standalone example and will update.

**Comment by romani** [comment]
Try to put "Id" on Check in configuration to make sure you do report of violation from right Check instance, example https://github.com/checkstyle/checkstyle/blob/master/config/checkstyle_checks.xml#L332.

If you do not provide issue report as we demand during 7 days, I will close the issue. If you want do discuss your problem, please do post in our maillist.

**Comment by ham1** [comment]
I put an ID and got the following:
```
components/org/apache/jmeter/extractor/BeanShellPostProcessor.java:47:5: Return count is 2 (max allowed is 1). [ReturnCount10]
components/org/apache/jmeter/modifiers/SampleTimeout.java:115:5: Return count is 2 (max allowed is 1). [ReturnCount10]
components/org/apache/jmeter/visualizers/PropertyControlGui.java:94:5: Return count is 2 (max allowed is 1). [ReturnCount10]
components/org/apache/jmeter/visualizers/backend/influxdb/UdpMetricsSender.java:81:5: Return count is 2 (max allowed is 1). [ReturnCount10]
core/org/apache/jmeter/gui/UndoHistory.java:104:5: Return count is 3 (max allowed is 1). [ReturnCount10]
core/org/apache/jmeter/gui/UndoHistory.java:306:5: Return count is 2 (max allowed is 1). [ReturnCount10]
core/org/apache/jmeter/gui/action/Load.java:73:5: Return count is 2 (max allowed is 1). [ReturnCount10]
core/org/apache/jmeter/gui/action/Save.java:127:5: Return count is 4 (max allowed is 1). [ReturnCount10]
core/org/apache/jmeter/gui/action/SaveGraphics.java:94:5: Return count is 2 (max allowed is 1). [ReturnCount10]
core/org/apache/jmeter/gui/action/SelectTemplatesDialog.java:133:5: Return count is 2 (max allowed is 1). [ReturnCount10]
core/org/apache/jmeter/gui/util/MenuFactory.java:616:5: Return count is 12 (max allowed is 10). [ReturnCount10]
core/org/apache/jmeter/gui/util/PowerTableModel.java:158:5: Return count is 11 (max allowed is 10). [ReturnCount10]
core/org/apache/jmeter/reporters/ResultSaver.java:174:5: Return count is 2 (max allowed is 1). [ReturnCount10]
core/org/apache/jmeter/save/converters/SampleSaveConfigurationConverter.java:79:9: Return count is 13 (max allowed is 10). [ReturnCount10]
core/org/apache/jmeter/util/XPathUtil.java:416:5: Return count is 4 (max allowed is 1). [ReturnCount10]
jorphan/org/apache/commons/cli/avalon/CLArgsParser.java:360:5: Return count is 2 (max allowed is 1). [ReturnCount10]
jorphan/org/apache/commons/cli/avalon/CLArgsParser.java:531:5: Return count is 2 (max allowed is 1). [ReturnCount10]
jorphan/org/apache/jorphan/gui/ComponentUtil.java:43:5: Return count is 2 (max allowed is 1). [ReturnCount10]
jorphan/org/apache/jorphan/gui/RateRenderer.java:41:5: Return count is 2 (max allowed is 1). [ReturnCount10]
jorphan/org/apache/jorphan/util/Converter.java:44:5: Return count is 13 (max allowed is 10). [ReturnCount10]
protocol/http/org/apache/jmeter/protocol/http/config/gui/UrlConfigGui.java:465:9: Return count is 4 (max allowed is 1). [ReturnCount10]
protocol/http/org/apache/jmeter/protocol/http/control/CacheManager.java:229:5: Return count is 2 (max allowed is 1). [ReturnCount10]
protocol/http/org/apache/jmeter/protocol/http/control/HttpMirrorServer.java:200:5: Return count is 2 (max allowed is 1). [ReturnCount10]
protocol/http/org/apache/jmeter/protocol/http/modifier/RegExUserParameters.java:58:5: Return count is 2 (max allowed is 1). [ReturnCount10]
protocol/http/org/apache/jmeter/protocol/http/modifier/URLRewritingModifier.java:70:5: Return count is 2 (max allowed is 1). [ReturnCount10]
protocol/http/org/apache/jmeter/protocol/http/proxy/HttpRequestHdr.java:181:5: Return count is 2 (max allowed is 1). [ReturnCount10]
```
I've tried to take one of the files which had "(max allowed is 1). [ReturnCount10]" and reproduce as required but as none of them are standalone files I've not had the time to make it work. If you have any time please see https://github.com/apache/jmeter/pull/345

**Comment by rnveach** [comment]
@ham1 
> `<module name="ReturnCount">`
> `<property name="max" value="10"/>`
> `</module>`
> I'd not expect "Return count is 2 (max allowed is 1)" when I've set it to 10.

This doesn't look like a false positive. Check has properties `max` and `maxForVoid`. You only specified `max` so `maxForVoid` is still using the default value of 1. If you want all returns to be 10, you must configure both properties.
http://checkstyle.sourceforge.net/config_coding.html#ReturnCount

> protocol/http/org/apache/jmeter/protocol/http/proxy/HttpRequestHdr.java:181:5: Return count is 2 (max allowed is 1). [ReturnCount10]

https://github.com/apache/jmeter/blob/69ed120d15615677c6074fb257baa20458fc560a/src/protocol/http/org/apache/jmeter/protocol/http/proxy/HttpRequestHdr.java#L181
This is a void method and I counted 2.

> jorphan/org/apache/jorphan/util/Converter.java:44:5: Return count is 13 (max allowed is 10). [ReturnCount10]

https://github.com/apache/jmeter/blob/e33bbb73fc343cfa187f49f6662904e4fda6d11e/src/jorphan/org/apache/jorphan/util/Converter.java#L44
This is a non-void method and I counted 13 returns.

I see no false positives here.

**Comment by ham1** [comment]
Ah, thank you, I missed that in the docs. Thank you for replying and sorting out a PEBKAC issue :)

**Comment by ham1** [comment]
P.S. is it worth changing the error message for void and non-void? I probably would have been able to solve this issue on my own had the error message been different e.g.
"Return count is 2 (maxVoid allowed is 1). [ReturnCount10]"
or
"Void return count is 2 (max allowed is 1). [ReturnCount10]"

**Comment by romani** [comment]
@ham1 , new issue is created - #5306 .
